Mathlib/Algebra/Algebra/Bilinear.lean

```diff
@@ -22,6 +22,19 @@ open TensorProduct Module
 
 namespace LinearMap
 
+/-- A version of `LinearMap.restrictScalars` for bilinear maps. -/
+@[simps!]
+def restrictScalars₂
+    (R : Type*) {A M N P : Type*}
+    [CommSemiring R] [CommSemiring A]
+    [AddCommMonoid M] [AddCommMonoid N] [AddCommMonoid P]
+    [Algebra R A]
+    [Module R M] [Module R N] [Module R P]
+    [Module A M] [Module A N] [Module A P]
+    [IsScalarTower R A M] [IsScalarTower R A N] [IsScalarTower R A P]
```

Comment:
Could this share some variables with the rest of the file?

Reply:
I found doing so rather difficult, there are too many typeclasses I don't want earlier in the file, and confusion over linear vs semilinear maps.